### PR TITLE
graphs/generators: add a generator for the Möbius ladder graph

### LINKED ISSUE #38083: Implementing a generator for Möbius ladder graph
### Problem Description

Currently, there is no existing implementation of the generator for [Möbius ladder graph](https://en.wikipedia.org/wiki/M%C3%B6bius_ladder).

### Proposed Solution

A simple implementation of the generator for Möbius ladder graph using the inspiration from Cycle graph as the base with the addition of the required 'rung' edges/ chords.

### Alternatives Considered

An alternative consideration would be to show it as a Möbius strip. 

For more information, please follow [the wikipedia article](https://en.wikipedia.org/wiki/M%C3%B6bius_ladder). You may find two different representation of Möbius ladder graph.

Currently, the second representation (the one with a single Cycle graph as the base) is being proposed to be implemented. Another alternative would be to consider the first representation.

### Additional Information

This implementation is a part of the project: [link](https://summerofcode.withgoogle.com/programs/2024/projects/iY5gx4Ls).

cc: @dcoudert 

### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### PR DESCRIPTION
**Pull request by janmenjayap**
This PR adds a generator for Möbius ladder graph.

This PR introduces a new static method, specifically `MoebiusLadderGraph`, that considers a non negative integer *n* as the input and returns the Möbius ladder graph of order *2n*. This also includes the required documentation concerning the same. 

It adds a new feature, that is $-$ a generator for Möbius ladder graph. 

Fixes #38083.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

Nothing as of now (up to my knowledge).

cc: @dcoudert .

### CODE REVIEW

**Review by dcoudert — COMMENTED** [review]
LGTM but you should add that this graph is also the circulant graph `graphs.CirculantGraph(2 * n, [1, n])`.

**Comment by Omegaconstant** [comment]
In the following line at line number 1164
For more information See `GraphGenerators.CirculantGraph()`,

it will be great if `GraphGenerators.CirculantGraph()` could be made a hyperlink that leads to [`GraphGenerators.CirculantGraph()`](https://doc.sagemath.org/html/en/reference/graphs/sage/graphs/graph_generators.html#sage.graphs.graph_generators.GraphGenerators.CirculantGraph).

Could you please mention how to do that?
Thank you.

File: src/sage/graphs/generators/basic.py [hunk 1]
```diff
@@ -1160,6 +1160,10 @@ def MoebiusLadderGraph(n):
     antipodal pairs of vertices. See the :wikipedia:`Möbius_ladder` for more
     information.
 
+    Note that :mod:`MoebiusLadderGraph(n)` is precisely the same graph as the
+    circulant graph :mod:`graphs.CirculantGraph(2 * n, [1, n])`. For more information
```
Comment (dcoudert) [inline src/sage/graphs/generators/basic.py]: use instead 
```
``graphs.CirculantGraph(2 * n, [1, n])`` 
```
you cannot make it a link.

Comment (dcoudert) [reply src/sage/graphs/generators/basic.py]: don't forget this change

File: src/sage/graphs/generators/basic.py [hunk 2]
```diff
@@ -1149,6 +1149,71 @@ def LadderGraph(n):
...
+    A Möbius ladder graph is a ladder graph that is connected at the ends with
+    a single twist, i.e.: a ladder bent around so that top meets bottom with a
+    single twist. Alternatively, it can be described as a single cycle graph (of
+    order `2 * n`) with the addition of edges (called 'rungs') connecting the
+    antipodal pairs of vertices. See the :wikipedia:`Möbius_ladder` for more
+    information.
+
+    Note that :mod:`MoebiusLadderGraph(n)` is precisely the same graph as the
+    circulant graph :mod:`graphs.CirculantGraph(2 * n, [1, n])`. For more information,
+    see :meth:`GraphGenerators.CirculantGraph`.
```
Comment (dcoudert) [inline src/sage/graphs/generators/basic.py]: I'm always confused with these links, but you can try: 
```
:meth:`~sage.graphs.graph_generators.GraphGenerators.CirculantGraph`.
```

File: src/sage/graphs/generators/basic.py [hunk 3]
```diff
@@ -1149,6 +1149,89 @@ def LadderGraph(n):
...
+    counterclockwise manner.
+
+    INPUT:
+
+    - ``n`` -- a non negative integer; number of nodes is `2n`
+
+    OUTPUT:
+
+    - ``G`` -- a Möbius graph of order `2n`; note that a :class:`ValueError` is
+      returned if ``n < 0``
```
Comment (dcoudert) [inline src/sage/graphs/generators/basic.py]: use ``` `n < 0` ```

File: src/sage/graphs/generators/basic.py [hunk 4]
```diff
@@ -1149,6 +1149,89 @@ def LadderGraph(n):
...
+
+    EXAMPLES:
+
+    Construct and show a Möbius ladder graph with 26 nodes::
+
+        sage: g = graphs.MoebiusLadderGraph(13)
+        sage: g.show()  # long time  # needs sage.plot
```
Comment (dcoudert) [inline src/sage/graphs/generators/basic.py]: the `# needs sage.plots` should start at columns 88. This is a convention to improve readability.

File: src/sage/graphs/generators/basic.py [hunk 5]
```diff
@@ -1149,6 +1149,89 @@ def LadderGraph(n):
...
+
+    Create several Möbius ladder graphs in a Sage graphics array::
+
+        sage: g = []
+        sage: j = []
+        sage: for i in range(9):
+        ....:    k = graphs.MoebiusLadderGraph(i+3)
+        ....:    g.append(k)
+        sage: for i in range(3):  # needs sage.plot
```
Comment (dcoudert) [inline src/sage/graphs/generators/basic.py]: here you can use block scope for `# needs sage.plots`, that is:
```
sage: # needs sage.plots
sage: g = []
...
```

**Review by dcoudert — CHANGES_REQUESTED** [review]

File: src/sage/graphs/generators/basic.py (hunk 5 above)
Comment (janmenjayap) [reply src/sage/graphs/generators/basic.py]: Is it fine if for this one also we go with `# needs sage.plots` mentioned at column no. 88 like the other ones (similar to the way it is mentioned here: [`graphs.CircularLadderGraph()`](https://doc.sagemath.org/html/en/reference/graphs/sage/graphs/graph_generators.html#sage.graphs.graph_generators.GraphGenerators.CircularLadderGraph)?
In other words, at line no. 1195 we will have the following:
```
sage: for i in range(3):                                                        # needs sage.plot
```

---

PS: In the first commit, I mentioned each of  `# needs sage.plots` at column no. 88 analogously the way it is written here:  [`graphs.CircularLadderGraph()`](https://doc.sagemath.org/html/en/reference/graphs/sage/graphs/graph_generators.html#sage.graphs.graph_generators.GraphGenerators.CircularLadderGraph), but in the [general convention](https://doc.sagemath.org/html/en/developer/coding_basics.html#python-code-style), I found the following line pertaining to sage documentation style:


*Lines should be shorter than 80 characters. If in doubt, read [PEP8: Maximum Line Length](https://www.python.org/dev/peps/pep-0008/#maximum-line-length).*

That's the reason, I changed those. Anyways, thanks for the info. 😊👍

**Comment by Omegaconstant** [comment]
Also, is it fine if we itemize the EXAMPLES with bullets since there are two of them? 

That is, instead of 
```
    Construct and show a Möbius ladder graph with 26 nodes::
```
and
```
   Create several Möbius ladder graphs in a Sage graphics array::
```

we may have the following:
```
    - Construct and show a Möbius ladder graph with 26 nodes::
```
and
```
     - Create several Möbius ladder graphs in a Sage graphics array::
```
respectively.

File: src/sage/graphs/generators/basic.py (hunk 5 above)
Comment (dcoudert) [reply src/sage/graphs/generators/basic.py]: The `# needs ...` tag is special and we try to minimize the impact on readability. In general we place it at column 88.
When we have a block of doctests with multiple time `# needs sage.plots`, we can use block scope, that is start the block of doctests with `sage: # needs sage.plots` and we don't add extra spaces. See for instance [`graphs.CompleteBipartiteGraph()`](https://doc.sagemath.org/html/en/reference/graphs/sage/graphs/graph_generators.html#sage.graphs.graph_generators.GraphGenerators.CompleteBipartiteGraph)

**Comment by dcoudert** [comment]
> Also, is it fine if we itemize the EXAMPLES with bullets since there are two of them?

No, we don't need bullets here. See how the documentation looks like.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by Omegaconstant** [comment]
Could you please let me know why **Build & Test / test-long (src/sage/[g-o]*) (pull_request)** is failing and how to rerun it?

**Comment by dcoudert** [comment]
an unexpected error when running `sage -t --long --random-seed=286735480429121101562228604801325644303 src/sage/numerical/linear_tensor_element.pyx`. I cannot reproduce on Fedora 39 and macOS. No need to rerun it I think (and I don't know how to do that anyway).

**Pull request merged**